Istio 1.14 release announcement, change-notes, and upgrade-notes (#11264)
istio/istio.io: content/en/news/releases/1.14.x/announcing-1.14/_index.md

---
title: Announcing Istio 1.14
linktitle: 1.14
subtitle: Major Update
description: Istio 1.14 release announcement.
publishdate: 2022-05-11
release: 1.14.0
skip_list: true
aliases:
- /news/announcing-1.14
- /news/announcing-1.14.0
---

We are pleased to announce the release of Istio 1.14!

{{< relnote >}}

This is the second Istio release of 2022. We would like to thank the entire Istio community
for helping to get Istio 1.14.0 published.
Special thanks are due to the release managers Lei Tang (Google) and Greg Hanson (solo.io),
and to Test & Release WG lead Eric Van Norman (IBM) for his help and guidance.

{{< tip >}}
Istio 1.14.0 is officially supported on Kubernetes versions `1.20` to `1.23`.
{{< /tip >}}

Here are some of the highlights of the release:

## Add a new approach for CA integration through the Envoy SDS API

This new feature adds a new approach for CA integration through the Envoy SDS API.

> For more information, check out the [documentation](/docs/ops/integrations/spire/).

Author: The doc at /docs/ops/integrations/spire/ is in the PR https://github.com/istio/istio.io/pull/11057, which is currently under review.

Author: The lint test failure for the broken link to /docs/ops/integrations/spire/ is expected because https://github.com/istio/istio.io/pull/11057 has not been merged yet.

Reviewer: Hey @lei-tang, just a friendly reminder that there is also supporting deployment example for this at https://github.com/istio/istio/pull/37948

> ## Add the auto-sni support

Author: @kfaseela Do you have a user guide (e.g., a link on the istio.io/docs) for this feature?

Reviewer 2: It's part of https://github.com/istio/istio.io/pull/11057 which hasn't merged yet.

Author: @ericvn Correct me if I am wrong. I think the #11057 is for the new feature of CA integration through the Envoy SDS API.
My first comment is about the user guide for the auto-sni feature.

Reviewer 2: You are correct.

Reviewer 3: Hello @lei-tang ,  this was a small enhancement to the existing DestinationRule logic, so I have not really prepared any documentation changes for the same. May be I can enhance the sni section at https://istio.io/latest/docs/reference/config/networking/destination-rule/ with additional description of the new feature flag, let me know.

Author: Your plan sounds good to me. After you complete the documentation, please let me know which link you would like to include in this PR. Thanks!

Reviewer 3: Ok, the documentation change for auto_sni is here:  https://github.com/istio/api/pull/2338
and the page that would get updated will be : https://istio.io/latest/docs/reference/config/networking/destination-rule/

Author: Thanks! I have added the link https://istio.io/latest/docs/reference/config/networking/destination-rule/ to the PR.

Reviewer 3: @ericvn May be you could get me a review on the above docs PR?

Author: https://github.com/istio/api/pull/2338 has been merged.

---
title: Announcing Istio 1.14
linktitle: 1.14
subtitle: Major Update
description: Istio 1.14 release announcement.
publishdate: 2022-05-11
release: 1.14.0
skip_list: true
aliases:
- /news/announcing-1.14
- /news/announcing-1.14.0
---

We are pleased to announce the release of Istio 1.14!

{{< relnote >}}

This is the second Istio release of 2022. We would like to thank the entire Istio community
for helping to get Istio 1.14.0 published.
Special thanks are due to the release managers Lei Tang (Google) and Greg Hanson (solo.io),
and to Test & Release WG lead Eric Van Norman (IBM) for his help and guidance.

{{< tip >}}

> Istio 1.14.0 is officially supported on Kubernetes versions `1.20` to `1.23`.

Author: Done.

{{< /tip >}}

Here are some of the highlights of the release:

## Support for the SPIRE runtime

SPIRE is a production-ready implementation of the SPIFFE specification, that offers
pluggable multi-factor attestation and SPIFFE federation. We've made changes in the way
we integrate with external Certificate Authorities, using the Envoy SDS API, to enable
support for SPIRE. Thanks to the team at HP Enterprise for contributing this work!

For more information, check out the [documentation](/docs/ops/integrations/spire/).

> ## Add the auto-sni support

Reviewer 2: Maybe remove `the`.

Author: Done.

Some servers require SNI be included in a request. This new feature configures SNI automatically.
For more information, check out the [documentation](/docs/reference/config/networking/destination-rule/).

> ## Add the support of configuring the TLS version for Istio workloads

Reviewer 2: Doesn't sound quit right. Maybe
```suggestion
## Add support for configuring the TLS version for Istio workloads
```

Author: Done.

TLS version is important for security. This new feature adds
the support of configuring the minimum TLS version for Istio workloads.
For more information, check out the [documentation](/docs/tasks/security/tls-configuration/workload-min-tls-version/).

## Telemetry improvements

The [Telemetry API](/docs/tasks/observability/telemetry/) has undergone a number of improvements,
including support for OpenTelemetry access logging, filtering based on WorkloadMode, and more.

## Upgrading to 1.14

> When you upgrade, we would like to hear from you! Please take a few minutes to respond to a brief [survey](https://forms.gle/pzWZpAvMVBecaQ9h9) to let us know how we’re doing.

Reviewer 2: Do we need to talk to Mitch about a new survey and link?

Author: @therealmitchconnors Do you have a new survey link?

Author: Based on the new survey link for 1.14 in https://github.com/istio/istio/pull/38790, the survey link has been updated.

release: 1.14.0
skip_list: true
aliases:
- /news/announcing-1.14
- /news/announcing-1.14.0
---

We are pleased to announce the release of Istio 1.14!

{{< relnote >}}

This is the second Istio release of 2022. We would like to thank the entire Istio community
for helping to get Istio 1.14.0 published.
Special thanks are due to the release managers Lei Tang (Google) and Greg Hanson (Solo.io),
and to Test & Release WG lead Eric Van Norman (IBM) for his help and guidance.

{{< tip >}}
Istio 1.14.0 is officially supported on Kubernetes versions `1.21` to `1.24`.
{{< /tip >}}

Here are some of the highlights of the release:

## Support for the SPIRE runtime

SPIRE is a production-ready implementation of the SPIFFE specification, that offers
pluggable multi-factor attestation and SPIFFE federation. We've made changes in the way
we integrate with external Certificate Authorities, using the Envoy SDS API, to enable
support for SPIRE. Thanks to the team at HP Enterprise for contributing this work!

SPIRE enables the introduction of strongly attested identities through the use of a combination
of different attestation mechanisms. It provides a variety of node and workload attestors out

> of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker, and, through a plugin

Reviewer 2: Remove commas

```suggestion
of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker and through a plugin
```

Author: Removed.

oriented architecture, it also enables the use of custom attestors.
The project has a pluggable integration with custom Key Management Systems for
storing the CA private keys, and enables integration with existing PKIs through the Upstream Certificate Authority plugin.
SPIRE implements SPIFFE Federation, enabling workloads to trust peers in a different trust domain, in
a configurable and flexible way through the Federation API.

> For more information, check out the [documentation](/docs/ops/integrations/spire/).

Reviewer 4: ```suggestion
For more information, check out the [documentation](/docs/ops/integrations/spire/) and this [livestream](https://www.youtube.com/watch?v=WOPoNqfrhb4). 
```

## Add auto-sni support

> Some servers require SNI be included in a request. This new feature configures SNI automatically.

Reviewer 4: ```suggestion
Some servers require SNI be included in a request. This new feature configures SNI automatically without users manually configure it or use an EnvoyFilter resource.
```

Author: Done.

> For more information, check out the [documentation](/docs/reference/config/networking/destination-rule/).

Reviewer 4: I checked, the doc doesn't say auto sni, may be best to refer to the PR:  https://github.com/istio/istio/pull/38604

Author: @kfaseela Can you take a look at the comment above?

Reviewer 3: @linsun : This PR was merged for clarifying the SNI behavior, hence pointing to the doc, not sure why it is not showing up? https://github.com/istio/api/pull/2338

Author: Based on the comments and offline discussions, the text has been updated to refer to the PRs.

## Add support for configuring the TLS version for Istio workloads

TLS version is important for security. This new feature adds
support for configuring the minimum TLS version for Istio workloads.

> For more information, check out the [documentation](/docs/tasks/security/tls-configuration/workload-min-tls-version/).

Reviewer 4: Is this link active?  I 'm getting 404 - https://istio.io/latest/docs/tasks/security/authentication/tls-configuration/workload-min-tls-version/

Author: The link is not at https://istio.io/latest/docs/tasks/security/authentication/tls-configuration/workload-min-tls-version/.
You can view the preliminary version at https://preliminary.istio.io/latest/docs/tasks/security/tls-configuration/workload-min-tls-version/

Reviewer 4: Perfect, thank you!

Istio 1.14.0 is officially supported on Kubernetes versions `1.21` to `1.24`.
{{< /tip >}}

Here are some of the highlights of the release:

## Support for the SPIRE runtime

SPIRE is a production-ready implementation of the SPIFFE specification, that offers
pluggable multi-factor attestation and SPIFFE federation. We've made changes in the way
we integrate with external Certificate Authorities, using the Envoy SDS API, to enable
support for SPIRE. Thanks to the team at HP Enterprise for contributing this work!

SPIRE enables the introduction of strongly attested identities through the use of a combination
of different attestation mechanisms. SPIRE provides a variety of node and workload attestors out
of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker, and, through a plugin
oriented architecture, it also enables the use of custom attestors.
SPIRE provides a pluggable integration with custom Key Management Systems, e.g. AWS KMS, for
storing the CA private keys.
SPIRE enables the integration with existing PKIs through the Upstream Certificate Authority plugin.
SPIRE implements SPIFFE Federation, enabling workloads to trust peers in a different trust domain, in

> a configurable and flexible way through the Federation API.

Reviewer 5: TODO rewrite "SPIRE is, SPIRE does, SPIRE etc" into a better paragraph

Author: @craigbox If you have suggestions on improving this paragraph, please let us know. We plan to release 1.14 around May 24.

> SPIRE enables the introduction of strongly attested identities through the use of a combination
> of different attestation mechanisms. SPIRE provides a variety of node and workload attestors out
> of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker, and, through a plugin
> oriented architecture, it also enables the use of custom attestors.
> SPIRE provides a pluggable integration with custom Key Management Systems, e.g. AWS KMS, for
> storing the CA private keys.
> SPIRE enables the integration with existing PKIs through the Upstream Certificate Authority plugin.
> SPIRE implements SPIFFE Federation, enabling workloads to trust peers in a different trust domain, in
> a configurable and flexible way through the Federation API.

Reviewer 5: ```suggestion
SPIRE enables the introduction of strongly attested identities through the use of a combination
of different attestation mechanisms. It provides a variety of node and workload attestors out
of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker, and, through a plugin
oriented architecture, it also enables the use of custom attestors.
The project has a pluggable integration with custom Key Management Systems for
storing the CA private keys, and enables integration with existing PKIs through the Upstream Certificate Authority plugin.
SPIRE implements SPIFFE Federation, enabling workloads to trust peers in a different trust domain, in
a configurable and flexible way through the Federation API.
```

Author: The edits look great and have been merged.

For more information, check out the [documentation](/docs/ops/integrations/spire/).

## Add auto-sni support

Some servers require SNI be included in a request. This new feature configures SNI automatically.
For more information, check out the [documentation](/docs/reference/config/networking/destination-rule/).

## Add support for configuring the TLS version for Istio workloads

TLS version is important for security. This new feature adds

> the support of configuring the minimum TLS version for Istio workloads.

Author: Done.

## Upgrading to 1.14

When you upgrade, we would like to hear from you! Please take a few minutes to respond to a brief [survey](https://forms.gle/yEtCbt45FZ3VoDT5A) to let us know how we’re doing.

You can also join the conversation at [Discuss Istio](https://discuss.istio.io/), or join our [Slack workspace](https://slack.istio.io/).
Would you like to contribute directly to Istio? Find and join one of our [Working Groups](https://github.com/istio/community/blob/master/WORKING-GROUPS.md) and help us improve.

## IstioCon wrap up

IstioCon 2022, the second edition of the project’s conference, took place Apr 25-29. We had almost 4,000 registered
participants, with a 4.5/5 satisfaction score. The conference was held in English and Chinese, with people
joining from 120 countries all over the world. During April 2022, the month of the conference, 81% of users
on istio.io were first time users. We will be sharing a more detailed report of the event on [events.istio.io](https://events.istio.io).

## CNCF wrap up

We're so pleased at the response to our announcement that [Istio has been proposed to the CNCF](/blog/2022/istio-has-applied-to-join-the-cncf/).

> We're hard at work at our application, and hope to have more to share in the coming months!

Author: Done.

{{< /tip >}}

Here are some of the highlights of the release:

## Support for the SPIRE runtime

SPIRE is a production-ready implementation of the SPIFFE specification, that offers
pluggable multi-factor attestation and SPIFFE federation. We've made changes in the way
we integrate with external Certificate Authorities, using the Envoy SDS API, to enable
support for SPIRE. Thanks to the team at HP Enterprise for contributing this work!

SPIRE enables the introduction of strongly attested identities through the use of a combination
of different attestation mechanisms. It provides a variety of node and workload attestors out
of the box for workloads running in Kubernetes, AWS, GCP, Azure, Docker and through a plugin
oriented architecture, it also enables the use of custom attestors.
The project has a pluggable integration with custom Key Management Systems for
storing the CA private keys, and enables integration with existing PKIs through the Upstream Certificate Authority plugin.
SPIRE implements SPIFFE Federation, enabling workloads to trust peers in a different trust domain, in
a configurable and flexible way through the Federation API.

> For more information, check out the [documentation](/docs/ops/integrations/spire/) and this [`livestream`](https://www.youtube.com/watch?v=WOPoNqfrhb4).

Reviewer 5: I am unsure if we should have a vendor link in here, and I definitely don't agree with putting back-ticks around the word.  It's not live any more, so maybe just 'video', and clarify that it's a Solo livestream not an Istio one?

@linsun

Reviewer 4: ```suggestion
For more information, check out the [documentation](/docs/ops/integrations/spire/) and this [`video`](https://www.youtube.com/watch?v=WOPoNqfrhb4).
```

Reviewer 4: @craigbox good point it is not live anymore. Can you check if the suggestion is acceptable?
